### DISCUSSION TITLE
Initial SOC

### DISCUSSION THREAD

**Discussion by HamidHamed1990** [pybamm-team/PyBaMM#3071] [Q&A]
How can I chnage the initial state of charge of the postivite/negative electrodes in the DFN model?

**Comment by brosaplanella** [comment]
When you solve a simulation you can use the argument `initial_soc`, see https://docs.pybamm.org/en/latest/source/api/simulation.html#pybamm.Simulation.solve

**Reply by HamidHamed1990** [reply]
Dear Ferran, thank you very much for the reply. I am curious about how to adjust the initial SOC of the cathode and anode separately. I don't know how adjusting initial_soc is related to each electrode SOC.

**Reply by brosaplanella** [reply]
The SoC only makes sense as a global quantity for the battery. If you want to adjust the initial stoichiometry of each electrode separately, you can update the initial concentrations in the parameter values.

**Reply by HamidHamed1990** [reply]
the full cell SOC relates to both cathode and Anode SOCs, so by defining one global SoC of battery, the individual electrodes' SOC cannot be determined. 
Thanks for the tip on 'initial concentrations'.

**Reply by Abhinav-Kshirsagar** [reply]
Dear @brosaplanella ,

This insight about stiochiometry being related to initial concentrations is really useful.
I was trying to figure out how could I directly vary the stoichiometry but this reply seems to be useful.

But i still fail to understand the corelation between the stoichiometry and initial concentration. When I set 
"Initial concentration in negative electrode [mol.m-3]": 37500,
"Initial concentration in positive electrode [mol.m-3]": 7750, 

What's the relation  between these values and stoichiometries of 0 and 1? or rather how to establish a relation? 
Could you please guide me in this case?

Thanks in advance for your help.

**Reply by brosaplanella** [reply]
PyBaMM uses an `eSOH` model (see https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/models/full_battery_models/lithium_ion/electrode_soh.py) to compute the SoC from the concentrations and the maximum and minimum voltages. The approach is explained in Section 3.2.1 of [Mohtat et al (2019)](https://www.sciencedirect.com/science/article/pii/S0378775319303593?casa_token=6XOidV0vjgAAAAAA:eDILuGOioUM6xq2vFb66wh64UcV6wW09tu9026cIWr5MDGlygcRocIWiHyhfHmpixXuB9-eJyeI).

**Reply by Abhinav-Kshirsagar** [reply]
Noted. Thanks!

**Comment by HamidHamed1990** [comment]
Dear Ferran,
I am also trying to introduce my own OCV data which is in the form of a csv file. I am using the following code:
```python3
import pybamm
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd


file_path = (
    "C:\\Users\\lucp9593\\Anaconda3\\Lib\\site-packages\\pybamm\\input\\parameters\\"
    "lithium_ion\\data/nmc_LGM50_ocp_Chen2020.csv"
)

model = pybamm.lithium_ion.DFN()
params = pybamm.ParameterValues("Chen2020")
ocp_NMC = pd.read_csv(file_path, comment="#", header=None).to_numpy()
ocp_NMC_interpol = pybamm.Interpolant(ocp_NMC[:, 0], ocp_NMC[:, 1], pybamm.t)
params["Positive electrode OCP [V]"] = ocp_NMC_interpol
solver = pybamm.CasadiSolver(mode="fast")
sim = pybamm.Simulation(model, parameter_values=params, solver=solver)
sim.solve([0, 3600])
sim.plot()
```

But I get this error: Solver failed because the following interpolation bounds were exceeded: ["Interpolant 'interpolating_function' lower bound"]. You may need to provide additional interpolation points outside these bounds.
May I ask how to solve this issue?

**Reply by brosaplanella** [reply]
Note that for the OCP, the interpolant variable should not be time but stoichiometry. You can find how to do that in `OKane2022.py`:
https://github.com/pybamm-team/PyBaMM/blob/5e28bb9a316f394282b40c8a56b285e85d987b77/pybamm/input/parameters/lithium_ion/OKane2022.py#L495-L497

**Reply by HamidHamed1990** [reply]
Thanks again.
